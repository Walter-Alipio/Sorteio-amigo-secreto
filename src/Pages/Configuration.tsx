import styled from 'styled-components';
import Card from '../components/Card/Card';
import Footer from '../components/Footer/Footer';
import Form from '../components/Form/Form';
import ParticipantsList from '../components/ParticipantsList/ParticipantsList';

const Container = styled.section`
	width: 100%;
	text-align: center;
	max-width: 900px;
`;

export const Title = styled.h2`
	color: rgba(75, 105, 253, 0.99);
	font-weight: 600;
	font-size: 1.25rem;
	padding-bottom: 4rem;
	text-align: center;

	@media (min-width: 800px) {
		font-size: 2rem;
	}
`;

const Configuration = () => {
	return (
		<Card>
			<Container>
				<Title>Vamos come√ßar!</Title>
				<Form />
				<ParticipantsList />
				<Footer />
			</Container>
		</Card>
	);
};

export default Configuration;
