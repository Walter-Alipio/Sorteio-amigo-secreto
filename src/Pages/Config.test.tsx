import { render } from '@testing-library/react';
import React from 'react';
import { RecoilRoot } from 'recoil';
import Configuration from './Configuration';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => {
	return {
		useNavigate: () => mockNavigate,
	};
});

describe('A página de configuração', () => {
	test('Deve ser renderizada corretamente', () => {
		const { container } = render(
			<RecoilRoot>
				<Configuration />
			</RecoilRoot>
		);

		//quando o teste é executado a primeira vez, cria uma imagem base do html que servira para comparação
		expect(container).toMatchSnapshot();
	});
});
