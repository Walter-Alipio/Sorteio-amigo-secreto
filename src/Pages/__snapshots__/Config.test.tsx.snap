// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A página de configuração Deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="sc-bczRLJ jsjhNL"
  >
    <section
      class="sc-papXJ gsubeQ"
    >
      <h2
        class="sc-jqUVSM eKIcEx"
      >
        Vamos começar!
      </h2>
      <form>
        <div
          class="sc-eCYdqJ fPQdnt"
        >
          <input
            class="sc-jSMfEi gdPKdK"
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            class="sc-jSMfEi sc-gKXOVf gdPKdK fFBcTU"
            disabled=""
          >
            Adicionar
          </button>
        </div>
        
      </form>
      <ul />
      <footer
        class="sc-gsnTZi bKhtBK"
      >
        <button
          class="sc-dkzDqf kzBkPq"
          disabled=""
        >
          Iniciar brincadeira
        </button>
        <img
          alt="Sacolas de compras"
          class="sc-hKMtZM gvFfar"
          src="/imagens/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
